pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -88,6 +88,10 @@ type Manager interface {
 
 	// GetAllocatableCPUs returns the assignable (not allocated) CPUs
 	GetAllocatableCPUs() cpuset.CPUSet
+
+	// GetCPUAffinity returns cpuset which includes cpus from shared pools
+	// as well as exclusively allocated cpus
+	GetCPUAffinity(podUID, containerName string) cpuset.CPUSet
```

Comment:
Ah, I somehow missed that this was an internal `CPUManager` API. I still think it's a better name, but I somehow thought that it was being exposed all the way through to the podresources API.

And if it's not ... I guess there is now no way for a consumer of the podresources to get the [taskset](https://man7.org/linux/man-pages/man1/taskset.1.html) (i.e. `CPUAffinity`) of a container (unless it has exclusive CPUs and `GetExclusiveCPUs() == GetCPUAffinity()`).

Reply:
I think so. I'm a little worried about the fact that we no longer have a way of getting the `taskset` of a container through this API anymore, and I'm a little worried about breaking people that relied on these semantics with the existing call (without giving them a way to get that information at all anymore). What's your sense on this?

Reply:
`Taskset` of the container provides information about the CPUs that the container can run on but what it lacking is way to determine whether those CPUs are exclusively allocated to that container or not. This leaves ambiguity in the API and prevents us from meeting the original goals that motivated these changes ([concrete resources](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2403-pod-resources-allocatable-resources#topology-aware-scheduling) and [getAllocatableResources](https://github.com/kubernetes/enhancements/tree/master/keps/sig-node/2403-pod-resources-allocatable-resources#topology-aware-scheduling) endpoint) in the podresource API in the first place.

My take is that it is appropriate to not consider the shared CPUs when we are talking about concrete resources allocated to a container because the container is not its sole consumer (CPUs are shared with other containers). Also, from client's perspective:

1. If the CPUs are exclusively allocated CPUs we have that information (with the changes proposed in this PR)
2. If the CPUs are not exclusively allocated, CPUs are obtained from the shared pool and this information can be obtained by calling the `GetAllocatableResources` endpoint

So essentially the information is there if the client wants it!

Reply:
Yeah, I understand the problem this PR is addressing. I just want to make sure someone can still get the same information they used to from the previous call to `GetGPUs()` before this change (essentially the `taskset` of the container).

To get that same info after this change, they will need to:
1. Call `GetAllocatableResources` to get a list of all the allocatable CPUs
2. Call `GetCPUs` on *all* containers in the system
3. Subtract out all of the CPUs from the `GetCPUs` calls from the `GetAllocatableResources` call

My guess is that most people that even use this API are mostly curious about what exclusive CPUs they have, so maybe this isn't a real issue. I just don't want anyone to be surprised by this change.

Reply:
Point taken! I will make sure that this is explicitly captured in the documentation [PR](https://github.com/kubernetes/website/pull/29751) and no one is taken by surprise. I added a [commit](https://github.com/kubernetes/website/pull/29751/commits/daa58fa14ad6d05de1f0038b1f696d1b16286eac) to the documentation PR to take care of this. Does that look like a reasonable way to take care of this?